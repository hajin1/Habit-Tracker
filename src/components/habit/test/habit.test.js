import Habit from "../habit";
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe('Habit component', () => {
    const habit = { name: 'Habit', count: 5 };
    let HabitComponent;
    let onIncrement;
    let onDecrement;
    let onDelete;
    let onReset;

    beforeEach(() => {
        onIncrement = jest.fn();
        onDecrement = jest.fn();
        onDelete = jest.fn();
        onReset = jest.fn();
        HabitComponent = (
            <Habit habit={habit} onIncrement={onIncrement} onDecrement={onDecrement} onDelete={onDelete} onReset={onReset} />
        );
    });

    it('renders', () => {
        const component = renderer.create(HabitComponent);
        expect(component.toJSON()).toMatchSnapshot();
    });

    describe('Button Click', () => {
        beforeEach(() => {
            render(HabitComponent);
        });

        it('calls onIncrement when clicking "increment" button', () => {
            const button = screen.getByTitle('increase');
            userEvent.click(button);
            expect(onIncrement).toHaveBeenCalledWith(habit);
        });

        it('calls onDecrement when clicking "decrement" button', () => {
            const button = screen.getByTitle('decrease');
            userEvent.click(button);
            expect(onDecrement).toHaveBeenCalledWith(habit);
        });

        it('calls onDelete when clicking "delete" button', () => {
            const button = screen.getByTitle('delete');
            userEvent.click(button);
            expect(onDelete).toHaveBeenCalledWith(habit);
        });
    });

});