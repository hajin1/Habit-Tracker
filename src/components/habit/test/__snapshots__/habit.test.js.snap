// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habit component renders 1`] = `
<li
  className="habit"
>
  <span
    className="habit-name"
    data-testid="habit-name"
  >
    Habit
  </span>
  <span
    className="habit-count"
    data-testid="habit-count"
  >
    5
  </span>
  <button
    className="habit-button habit-increase"
    data-testid="habit-increase"
    onClick={[Function]}
    title="increase"
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <button
    className="habit-button habit-decrease"
    data-testid="habit-decrease"
    onClick={[Function]}
    title="decrease"
  >
    <i
      className="fas fa-minus-square"
    />
  </button>
  <button
    className="habit-button habit-delete"
    data-testid="habit-delete"
    onClick={[Function]}
    title="delete"
  >
    <i
      className="fas fa-trash"
    />
  </button>
</li>
`;
